astropy/utils/masked/core.py

```diff
@@ -1113,7 +1113,15 @@ def mean(self, axis=None, dtype=None, out=None, keepdims=False, *, where=True):
             axis=axis, dtype=dtype, out=out, keepdims=keepdims, where=where
         )
         n = np.add.reduce(where, axis=axis, keepdims=keepdims)
+
+        # catch the case when an axis is fully masked to prevent div by zero:
+        n = np.add.reduce(where, axis=axis, keepdims=keepdims)
+        neq0 = n == 0
+        n += neq0
         result /= n
+        # mask resulting values from fully-masked axes
```

Comment:
Maybe the comment should be changed to something like "# Correct fully-masked slice results to what is expected for 0/0 division." -- we're not touching the mask here!

Reply:
Much clearer, thanks.